src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/data/VisitResponseTest.kt

```diff
@@ -211,6 +227,34 @@ internal class VisitResponseTest {
         "0123456789",
       )
     }
+
+    @Test
+    internal fun `outcome is derived from dummy visitor record`() {
+      val theVisit =
+        visit.copy(
+          visitors = mutableListOf(
+            visitor,
+            outcomeVisitorRecord.copy(outcomeReason = VisitOutcomeReason("REFUSED", "Offender Refused Visit"))
+          )
+        )
+      val response = VisitResponse(theVisit)
+
+      assertThat(response.visitOutcome).isEqualTo(CodeDescription("REFUSED", "Offender Refused Visit"))
+    }
+
+    @Test
+    internal fun `outcome is null when not present in dummy visitor record`() {
+      val theVisit =
+        visit.copy(
+          visitors = mutableListOf(
+            visitor,
+            outcomeVisitorRecord.copy(outcomeReason = null)
+          )
+        )
+      val response = VisitResponse(theVisit)
+
+      assertThat(response.visitOutcome).isNull()
+    }
   }
 }
```

Comment:
hard to tell - but do you have a test with an outcome on a non visitor record just to check the dummy one is returned?

Reply:
Good point - all the tests currently have no outcomes on the real visitor records where as think they would be there. I'll add a quick unit test to show they are ignored